when I `make run` on the master branch, i get an unsurprising `403: Forbidden` from redcap.
when i `make run` on this branch, i get this stacktrace:
```
python3 ref_code_gen.py
Traceback (most recent call last):
  File "ref_code_gen.py", line 163, in <module>
    _script_io()
  File "ref_code_gen.py", line 161, in _script_io
    main(argv[:], environ.copy(), logging.basicConfig, build_opener())
  File "ref_code_gen.py", line 51, in main
    invite_code_type = argv[3]
IndexError: list index out of range
make: *** [Makefile:19: run] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -15,13 +15,19 @@ lint: $(SRCS)
 doctest:
 	$(PYTHON3) -m doctest $(SRCS)
 
-run: lint check doctest ref_code_gen.py
-	$(PYTHON3) ref_code_gen.py
+set_env_variable:
+	echo "Make sure to set REDCAP_API_TOKEN as env variable."
 
-integration_test: lint check doctest
-	REDCAP_API_TOKEN=$(REDCAP_API_TOKEN) $(PYTHON3) ref_code_gen.py 2 3
+run_test: lint check doctest set_env_variable ref_code_gen.py
+	REDCAP_API_TOKEN=$(REDCAP_API_TOKEN) $(PYTHON3) ref_code_gen.py 5 5 test
 
-integration_test_pdf: lint check doctest
+run_production: lint check doctest set_env_variable ref_code_gen.py
+	REDCAP_API_TOKEN=$(REDCAP_API_TOKEN) $(PYTHON3) ref_code_gen.py 1000 5 production
+
+integration_test: lint check doctest set_env_variable
+	REDCAP_API_TOKEN=$(REDCAP_API_TOKEN) $(PYTHON3) ref_code_gen.py 5 5 test
+
+integration_test_pdf: lint check doctest set_env_variable
 	REDCAP_API_TOKEN=$(REDCAP_API_TOKEN) $(PYTHON3) ds_status_sync.py --send-consent REDCAP_API_TOKEN
 
 debug: lint check

--- a/ref_code_gen.py
+++ b/ref_code_gen.py
@@ -104,6 +104,7 @@ class ReferralCode:
         crc = crc32(base.encode('utf-8'))
         digit = crc % 10
         out = f'{base}{digit}'
+        # uncomment following to understand the logic.
         # print(candidate, base_len, base, crc, digit, out)
         if out != candidate and len(candidate) > base_len:
             raise ValueError(candidate)

